### DISCUSSION TITLE
BuildMode Solid/Segments for regular shapes

### DISCUSSION THREAD

**Discussion by Boivin92** [godotengine/godot-proposals#5239] [2D]
I'm curious as to why the BuildMode option is only available for CollisionPolygons rather than all collision objects. I'd like to have a circle acting as a form of hamster-wheel but it seems I'd need to create a 64-sided "circle" by code in order to achieve that.

Is it something a casual programmer could contribute a PR to the engine or is it a terribly complex physics problem that no one wants to touch?

**Comment by Calinou** [comment]
For performance, primitive collision shapes aren't built out of several convex collision shapes. Instead, they use dedicated optimized codepaths.

I suggest implementing this with an add-on, so you can create circle collision shapes of arbitrary thickness.
